.github/workflows/build.yml

```diff
@@ -9,26 +9,36 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
     name: Build and tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+        java: [ 8, 11, 15, 17 ]
+    runs-on: ${{ matrix.os }}
     outputs:
       project_version: ${{ steps.get_project_version.outputs.project_version }}
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 8
+      - uses: actions/checkout@v3
+      - name: Set up JDK ${{ matrix.java }}
         uses: actions/setup-java@v2
         with:
-          java-version: '8'
+          java-version: ${{ matrix.java }}
           distribution: 'adopt'
           cache: 'gradle'
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Fix Gradle options on Java < 9
```

Comment:
the additional gradle options that I had to add to have the linter working are not compatible with versions of Java below 9 😠